gtk3 part doesn't install gcc

This issue is similar with #141, but is with the gtk3 part.

[It is possible one's Gtk3 snap does not directly require a C toolchain](https://github.com/Lin-Buo-Ren/snapcrafters-template-plus/blob/master/snap/snapcraft.yaml) and thus doesn't depend on GCC as `build-packages`, if the remote part requires one it must declare by itself.

```
'yad' has dependencies that need to be staged: desktop-gtk3                                             
Skipping pull desktop-gtk3 (already ran)
Building desktop-gtk3 
make FLAVOR=gtk3 -j4
gcc -Wall -O2 -o bindtextdomain.so -fPIC -shared ./../src/bindtextdomain.c -ldl
make: gcc: Command not found
Makefile:18: recipe for target 'desktop-launch' failed
make: *** [desktop-launch] Error 127
Failed to run 'make FLAVOR=gtk3 -j4' for 'desktop-gtk3': Exited with code 2.                            
Verify that the part is using the correct parameters and try again.                                     
Stopping local:snapcraft-grandfatherly-orogenetic-chet
```

--- a/snapcraft.yaml
+++ b/snapcraft.yaml
@@ -160,6 +160,7 @@ parts:
     plugin: make
     make-parameters: ["FLAVOR=gtk2"]
     build-packages:
+      - build-essential
       - libgtk2.0-dev
     stage-packages:
       - libxkbcommon0  # XKB_CONFIG_ROOT
@@ -185,6 +186,7 @@ parts:
     plugin: make
     make-parameters: ["FLAVOR=gtk3"]
     build-packages:
+      - build-essential
       - libgtk-3-dev
     stage-packages:
       - libxkbcommon0  # XKB_CONFIG_ROOT
@@ -263,6 +265,7 @@ parts:
     plugin: make
     make-parameters: ["FLAVOR=gtk3"]
     build-packages:
+      - build-essential
       - libgtk-3-dev
     override-build: |
       snapcraftctl build

